dynawo/sources/Models/CPP/ModelFrequency/ModelSignalN/DYNModelSignalN.cpp

```diff
@@ -0,0 +1,408 @@
+//
+// Copyright (c) 2015-2020, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+// This file is part of Dynawo, an hybrid C++/Modelica open source time domain
+// simulation tool for power systems.
+//
+
+/**
+ * @file  DYNModelSignalN.cpp
+ *
+ * @brief SignalN model implementation
+ *
+ * All generators of the network are connected to this model through their participation in the frequency
+ * regulation Alpha, which is a coefficient that can be equal to the nominal active power of the generator,
+ * its maximum active power, or its active power set point, through the total participation SignalN, which
+ * is calculated in this model and is equal to the sum of the Alpha, and through the signal N, which is common
+ * to all the generators in the same connected component and that changes the active power reference of the
+ * generators to balance the generation and the consumption. This signal N emulates a primary frequency regulation.
```

Comment:
Yes and no for the primary frequency regulation.
It is indeed instantaneous but depending on the coefficient you use, you can end up on a steady-state that is closer to the action of either primary or secondary frequency regulation. I would delete this sentence.